test/e2e/resource_test.go

```diff
@@ -183,6 +185,75 @@ var _ = ginkgo.Describe("[resource-status collection] resource status collection
 		})
 	})
 
+	ginkgo.Context("ServiceStatus collection testing", func() {
+		var serviceNamespace, serviceName string
+		var service *corev1.Service
+
+		ginkgo.BeforeEach(func() {
+			policyNamespace = testNamespace
+			policyName = deploymentNamePrefix + rand.String(RandomStrLength)
+			serviceNamespace = testNamespace
+			serviceName = policyName
+
+			service = helper.NewService(serviceNamespace, serviceName)
+			service.Spec.Type = corev1.ServiceTypeNodePort
+			policy = helper.NewPropagationPolicy(policyNamespace, policyName, []policyv1alpha1.ResourceSelector{
+				{
+					APIVersion: service.APIVersion,
+					Kind:       service.Kind,
+					Name:       service.Name,
+				},
+			}, policyv1alpha1.Placement{
+				ClusterAffinity: &policyv1alpha1.ClusterAffinity{
+					ClusterNames: framework.ClusterNames(),
+				},
+			})
+		})
+
+		ginkgo.BeforeEach(func() {
+			framework.CreateService(kubeClient, service)
+			ginkgo.DeferCleanup(func() {
+				framework.RemoveService(kubeClient, serviceNamespace, serviceName)
+			})
+		})
+
+		ginkgo.It("NodePort service apply status collection testing", func() {
+			nodePorts := sets.NewInt32()
+			// simulate the update of the service status in member clusters.
+			ginkgo.By("Update service status in member clusters", func() {
+
+				for _, clusterName := range framework.ClusterNames() {
+					clusterClient := framework.GetClusterClient(clusterName)
+					gomega.Expect(clusterClient).ShouldNot(gomega.BeNil())
+					gomega.Eventually(func(g gomega.Gomega) {
+						memberSvc, err := clusterClient.CoreV1().Services(serviceNamespace).Get(context.TODO(), serviceName, metav1.GetOptions{})
+						g.Expect(err).NotTo(gomega.HaveOccurred())
+						for _, servicePort := range memberSvc.Spec.Ports {
+							nodePorts.Insert(servicePort.NodePort)
+						}
+					}, pollTimeout, pollInterval).Should(gomega.Succeed())
+				}
+				// check service nodePort
+				gomega.Expect(nodePorts.Len() == 1).Should(gomega.BeTrue())
+			})
+			klog.Infof("svcNodePort: %v", nodePorts.List()[0])
+
+			ginkgo.By("check service ResourceBindings apply status ", func() {
+				gomega.Eventually(func(g gomega.Gomega) (bool, error) {
+					resourceBindingName := names.GenerateBindingName(service.Kind, service.Name)
+					resourceBinding, err := karmadaClient.WorkV1alpha2().ResourceBindings(serviceNamespace).Get(context.TODO(), resourceBindingName, metav1.GetOptions{})
+					g.Expect(err).NotTo(gomega.HaveOccurred())
+					statuses := sets.NewString()
+					for _, condition := range resourceBinding.Status.Conditions {
+						statuses.Insert(string(condition.Status))
+					}
+					gomega.Expect(statuses.Len() == 1).Should(gomega.BeTrue())
+					return gomega.Expect(statuses.List()[0] == "True").Should(gomega.BeTrue()), nil
+				}, pollTimeout, pollInterval).Should(gomega.Equal(true))
```

Comment:
We will collect the status of resources to the resource template(such as service resources), so we can check service status directly here.

Reply:
Can you help me find a sample?

Reply:
yes ,it is ,service of the Nodeport type do not collect status,therefore, we only judge according to the status of `resourcebinding`

Reply:
We can directly get the `FullyApplied` type condition of the resourcebinding status,

Reply:
> We can directly get the FullyApplied type condition of the resourcebinding status,

Can you help me find a sample?

Reply:
How about this:

```diff
--- a/test/e2e/resource_test.go
+++ b/test/e2e/resource_test.go
@@ -19,6 +19,7 @@ import (
        "k8s.io/klog/v2"

        policyv1alpha1 "github.com/karmada-io/karmada/pkg/apis/policy/v1alpha1"
+       workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
        "github.com/karmada-io/karmada/pkg/util/names"
        "github.com/karmada-io/karmada/test/e2e/framework"
        "github.com/karmada-io/karmada/test/helper"
@@ -239,17 +240,19 @@ var _ = ginkgo.Describe("[resource-status collection] resource status collection
                        klog.Infof("svcNodePort: %v", nodePorts.List()[0])

                        ginkgo.By("check service ResourceBindings apply status ", func() {
-                               gomega.Eventually(func(g gomega.Gomega) (bool, error) {
+                               gomega.Eventually(func(g gomega.Gomega) (metav1.ConditionStatus, error) {
                                        resourceBindingName := names.GenerateBindingName(service.Kind, service.Name)
                                        resourceBinding, err := karmadaClient.WorkV1alpha2().ResourceBindings(serviceNamespace).Get(context.TODO(), resourceBindingName, metav1.GetOptions{})
                                        g.Expect(err).NotTo(gomega.HaveOccurred())
-                                       statuses := sets.NewString()
+
+                                       var fullyAppliedStatus metav1.ConditionStatus
                                        for _, condition := range resourceBinding.Status.Conditions {
-                                               statuses.Insert(string(condition.Status))
+                                               if condition.Type == workv1alpha2.FullyApplied {
+                                                       fullyAppliedStatus = condition.Status
+                                               }
                                        }
-                                       gomega.Expect(statuses.Len() == 1).Should(gomega.BeTrue())
-                                       return gomega.Expect(statuses.List()[0] == "True").Should(gomega.BeTrue()), nil
-                               }, pollTimeout, pollInterval).Should(gomega.Equal(true))
+                                       return fullyAppliedStatus, nil
+                               }, pollTimeout, pollInterval).Should(gomega.Equal(metav1.ConditionTrue))
                        })
                })
        })
```

Reply:
nice